lib/functoria/filegen.ml

```diff
@@ -42,12 +43,15 @@ module Make (P : PROJECT) = struct
     | `Sexp -> Fmt.str ";; %s" line
     | `Opam | `Make -> Fmt.str "# %s" line
     | `OCaml -> Fmt.str "(* %s *)" line
+    | `OpamLock ->
+        Fmt.invalid_arg "Opam lockfiles are generated by opam-monorepo."
 
   let short_headers lang =
     match lang with
     | `Sexp -> Fmt.str ";; Generated by"
     | `Opam | `Make -> "# Generated by"
     | `OCaml -> "(* Generated by"
+    | `OpamLock -> "synopsis: \"duniverse generated"
```

Comment:
why does `headers` and `short_headers` disagree on what to do for `OpamLock`?

Reply:
Also: `OpamLock` doesn't seem to be used elsewhere?

Reply:
`headers` is used when Mirage generates files.
`short_headers` is used to identify files generated by Mirage.

For the opam lockfile, only the identification function (`short_headers`) was used to remove the file in the `clean` step. Now that most mirage files (include the lockfile) are stored together inside the `mirage/` folder, there's no need to identify `OpamLock` anymore so it can be removed